src/Commands/DrupalCommands.php

```diff
@@ -273,17 +273,34 @@ public function drushSetup(array $options = [
      * @command drupal:settings-setup
      *
      * @option root Drupal root.
+     * @option sites-subdir Drupal site subdirectory.
+     * @option settings-override-file Drupal site settings override filename.
      *
      * @param array $options
      *
      * @return \Robo\Collection\CollectionBuilder
      */
     public function settingsSetup(array $options = [
-      'root' => InputOption::VALUE_REQUIRED,
+        'root' => InputOption::VALUE_REQUIRED,
+        'sites-subdir' => InputOption::VALUE_REQUIRED,
+        'settings-override-file' => InputOption::VALUE_REQUIRED,
     ])
     {
+        $settings_file = $options['root'] . '/sites/' . $options['sites-subdir'] . '/settings.php';
+        $settings_override_filename = $options['settings-override-file'];
+        $settings_override_path = $options['root'] . '/sites/' . $options['sites-subdir'] . '/' . $settings_override_filename;
         return $this->collectionBuilder()->addTaskList([
-            $this->taskAppendConfiguration($options['root'].'/sites/default/default.settings.php', $this->getConfig())->setConfigKey('drupal.settings'),
+            $this->taskFilesystemStack()->copy($options['root'] . '/sites/default/default.settings.php', $settings_file, true),
```

Comment:
`$options['sites-subdir']` should replace `/default/` on these two lines above too.